src/server/context.ts

```diff
@@ -323,7 +323,7 @@ export class ServerContext {
       // Ex: /about/ -> /about
       const url = new URL(req.url);
       if (url.pathname.length > 1 && url.pathname.endsWith("/")) {
-        url.pathname = url.pathname.slice(0, -1);
+        url.pathname = url.pathname.replace(/\/+/g, '/').slice(0, -1);
```

Comment:
This replaces all slashes (not just the final ones. That behaviour is fine, but it should probably happen _before_ this `if` statement (because right now `/a///b/` normalizes to `/a/b`, but `/a///b` does not.

Can you also add a test case for this scenario?

Reply:
I fixed this, and added a test. Let me know if it needs anything else.